tests/promptflow/test_promptflow_model_export.py

```diff
@@ -0,0 +1,103 @@
+import json
+from pathlib import Path
+
+import pytest
+from promptflow import load_flow
+from promptflow._sdk.entities._flow import Flow
+from pyspark.sql import SparkSession
+
+import mlflow
+from mlflow import MlflowException
+from mlflow.deployments import PredictionsResponse
+from mlflow.pyfunc.scoring_server import CONTENT_TYPE_JSON
+
+from tests.helper_functions import pyfunc_serve_and_score_model
+
+
+@pytest.fixture(scope="module")
+def spark():
+    with SparkSession.builder.master("local[*]").getOrCreate() as s:
+        yield s
+
+
+def get_promptflow_example_model():
+    flow_path = Path(__file__).parent / "flow_with_additional_includes"
+    return load_flow(flow_path)
+
+
+def test_promptflow_log_and_load_model():
+    model = get_promptflow_example_model()
+    with mlflow.start_run():
+        logged_model = mlflow.promptflow.log_model(
+            model, "promptflow_model", input_example={"text": "Python Hello World!"}
+        )
+
+    loaded_model = mlflow.promptflow.load_model(logged_model.model_uri)
+
+    assert "promptflow" in logged_model.flavors
+    assert logged_model.signature is not None
+    assert str(logged_model.signature.inputs) == "['text': string]"
+    assert str(logged_model.signature.outputs) == "['output': string]"
+    assert isinstance(loaded_model, Flow)
+
+
+def test_log_model_with_config():
+    model = get_promptflow_example_model()
+    model_config = {"connection.provider": "local"}
+    with mlflow.start_run():
+        logged_model = mlflow.promptflow.log_model(
+            model, "promptflow_model", model_config=model_config
+        )
+
+    assert mlflow.pyfunc.FLAVOR_NAME in logged_model.flavors
+    assert mlflow.pyfunc.MODEL_CONFIG in logged_model.flavors[mlflow.pyfunc.FLAVOR_NAME]
+    logged_model_config = logged_model.flavors[mlflow.pyfunc.FLAVOR_NAME][
+        mlflow.pyfunc.MODEL_CONFIG
+    ]
+    assert logged_model_config == model_config
+
+
+def test_promptflow_model_predict(spark):
```

Comment:
Let's split this to multiple test cases. Bundling all cases into one prevents quick iteration, e.g. none of the checks until the last assertion need spark session.